### ISSUE TITLE
Removal of BufferedShuffleDataset

### ISSUE THREAD

**Issue by dvruette** [pytorch/pytorch#66116]
## 📚 Documentation

The removal of the `BufferedShuffleDataset` class from the `torch.utils.data` module isn't documented either in the release notes nor in the documentation page about `torch.utils.data`. The only way to find out that it has indeed been removed and actually replaced by `torch.utils.data.datapipes.iter.combinatorics.ShuffleIterDataPipe` is by finding #62310.

Since this is a breaking change and not trivial to resolve, either a deprecation notice should be added to the documentation of `BufferedShuffleDataset` [1] or a note about what replaced it should be added to the current documentation [2].

[1] https://pytorch.org/docs/1.8.1/data.html#torch.utils.data.BufferedShuffleDataset
[2] https://pytorch.org/docs/1.9.1/data.html
